### PR TITLE
Move compare_and_set_snapshot out of Scraper subclasses (#121)

### LINKED ISSUE #121: Move compare_and_set_snapshot() into the parent Scraper class
Right now, all child classes have to implement this function, which is only different because of the name of the child snapshot class that must be referenced. This will involve adding functionality to utils/regions.py to retrieve a region's snapshot class.

### PR DESCRIPTION
**Pull request by ageiduschek**
Fixes #121 - Moves some duplicated code out of Snapshot base classes.

Note: When running the us_ny scraper locally on I'm seeing a ton of seemingly unrelated exceptions, but this change doesn't seem to have broken things more...

### CODE REVIEW

**Comment by jessex** [comment]
I'm pretty sure I know what the exceptions you saw locally are but if you send them over to me I can confirm. A branch I have in progress fixes them in passing if they're what I think they are. If so, I'll issue a quick PR with just that fix.

File: recidiviz/ingest/scraper.py [hunk 1]
```diff
@@ -85,6 +88,21 @@ def person_id_to_record_id(self, person_id):
         """
         pass
 
+
+    @abc.abstractmethod
+    def get_snapshot_class_name(self):
```
Comment (jessex) [inline recidiviz/ingest/scraper.py]: We can remove this in favor of `Region.get_snapshot_kind()`, which we can call directly from `compare_and_set_snapshot`.

File: recidiviz/ingest/scraper.py [hunk 2]
```diff
@@ -334,3 +352,90 @@ def iterate_docket_item(self, scrape_type):
...
+                current scrape.
+        Returns:
+            True if successful
+            False if datastore errors
+
+        """
+        new_snapshot = False
+
+        # pylint:disable=protected-access
+        snapshot_class = ndb.Model._kind_map[self.get_snapshot_class()]
```
Comment (jessex) [inline recidiviz/ingest/scraper.py]: Once we remove this in favor of `Region.get_snapshot_kind`, we should be able to annotate this method as a `@staticmethod`.

File: recidiviz/ingest/scraper.py [hunk 3]
```diff
@@ -334,3 +352,90 @@ def iterate_docket_item(self, scrape_type):
...
+                        setattr(snapshot, attribute, None)
+
+                elif attribute == "offense":
+                    # Offenses have to be treated differently -
+                    # setting them to None means an empty list or
+                    # tuple instead of None, and an unordered list of
+                    # them can't be compared to another unordered list
+                    # of them.
+                    #
+                    # TODO #123 instead of storing the entire list if
```
Comment (jessex) [inline recidiviz/ingest/scraper.py]: If we're going to _not_ do #123 as discussed offline, would you mind removing this TODO and closing that issue with a comment that explains why?

File: recidiviz/ingest/scraper.py (hunk 2 above)
Comment (jessex) [reply recidiviz/ingest/scraper.py]: Never mind. As discussed offline, you're right. We'd still be accessing `self.region` here.

File: recidiviz/ingest/us_ny/us_ny_scraper.py [hunk 4]
```diff
@@ -911,7 +911,7 @@ def store_record(self, person_details):
             record.offense = record_offenses
         record.race = person.race
         record.record_id = record_id
-        record.region = self.get_region()
+        record.region = self.get_region().region_code
```
Comment (jessex) [inline recidiviz/ingest/us_ny/us_ny_scraper.py]: Thanks for tossing in this fix, @ageiduschek! I'll close #143.

Comment (ageiduschek) [reply recidiviz/ingest/us_ny/us_ny_scraper.py]: Oh whooops I thought I had reverted this. Somehow that didn't make it up here...

**Pull request merged**